src/backend/strategies.rs

```diff
@@ -306,21 +306,23 @@ pub(crate) async fn run_strategy_task<'s>(
                 };
 
                 // Get the loaded identity nonce
-                let loaded_identity_lock = match app_state.refresh_identity(&sdk).await {
-                    Ok(lock) => lock,
-                    Err(e) => {
-                        error!("Failed to refresh identity: {:?}", e);
-                        return BackendEvent::StrategyError {
-                            strategy_name: strategy_name.clone(),
-                            error: format!("Failed to refresh identity: {:?}", e),
-                        };
-                    }
-                };
+                let mut loaded_identity_lock = app_state.loaded_identity.lock().await;
+                if loaded_identity_lock.is_some() {
+                    drop(loaded_identity_lock);
+            
+                    let _ = app_state.refresh_identity(&sdk).await;
```

Comment:
that is all weird, could `refresh_identity` do this check since it's already taking a lock inside of it?